File: 2012_2013_sujet_2_simulation_de_lois/discrete2.m
function res=discrete2(n,p)
m=length(p);
X=rand(1,n);
p2=cumsum(p);
res=X<p2(1);
for i=2:1:m
    res=res+i*(X>p2(i-1)&X<p2(i));
end
end

File: 2011_2012_sujet_2_simulation_de_lois/discrete2.py
import numpy as np


def discrete2(n, p):
    m = len(p)
    X = np.random.rand(n)
    p2 = np.cumsum(p)
    res = (X < p2[0]).astype(int)
    for i in range(1, m):
        res = res + (i + 1) * ((X > p2[i-1]) & (X < p2[i])).astype(int)
    return res
